app/records/views.py

```diff
@@ -1,36 +1,79 @@
-from app.ciim.exceptions import DoesNotExist
-from app.records.api import records_client
+from app.lib.api import ResourceNotFound
+from app.records.api import record_details_by_id, record_details_by_ref
+from app.records.labels import FIELD_LABELS, LEVEL_LABELS, NON_TNA_LEVEL_LABELS
 from django.http import Http404
+from django.template.loader import get_template
 from django.template.response import TemplateResponse
+from django.urls import reverse
+
+# TODO: Implement record_detail_by_reference once Rosetta has support
+# def record_detail_by_reference(request, reference):
+#     """
+#     View for rendering a record's details page.
+#     """
+#     template_name = "records/record_detail.html"
+#     context = {"field_labels": FIELD_LABELS, "level_labels": LEVEL_LABELS, "non_tna_level_labels": NON_TNA_LEVEL_LABELS}
+
+#     try:
+#         # record = record_details_by_ref(id=reference)
+#         record = record_details_by_id(id="D4664016")
+#     except ResourceNotFound:
+#         raise Http404
+#     except Exception:
+#         raise HttpResponseServerError
+
+#     context.update(
+#         record=record,
+#         canonical=reverse(
+#             "details-page-machine-readable", kwargs={"id": record.id}
+#         ),
+#     )
+
+#     if record.custom_record_type:
+#         if record.custom_record_type == "ARCHON":
+#             template_name = "records/archon_detail.html"
+#         if record.custom_record_type == "CREATORS":
+#             template_name = "records/creator_detail.html"
+
+#     return TemplateResponse(
+#         request=request, template=template_name, context=context
+#     )
 
 
 def record_detail_view(request, id):
     """
     View for rendering a record's details page.
     """
     template_name = "records/record_detail.html"
-    context = {}
-    page_type = "Record details page"
+    context: dict = {
+        "field_labels": FIELD_LABELS,
+        "level_labels": LEVEL_LABELS,
+        "non_tna_level_labels": NON_TNA_LEVEL_LABELS,
+    }
 
     try:
-        # for any record
-        record = records_client.get(id=id)
-
-        if record.custom_record_type and record.custom_record_type != "CAT":
-            # raise error for any other types ex ARCHON, CREATORS
-            # TODO: other types ex ARCHON, CREATORS, will have their own details page templates
-            raise DoesNotExist
-
-        page_title = f"Catalogue ID: {record.iaid}"
-    except DoesNotExist:
+        record = record_details_by_id(id=id)
+    except ResourceNotFound:
         raise Http404
+    except Exception as e:
+        context = {"exception_message": e}
+        return TemplateResponse(
+            request=request,
+            template="errors/server_error.html",
+            context=context,
+            status=502,
+        )
```

Comment:
The `TypeError at /catalogue/id/C99991/ exceptions must derive from BaseException` is resolved.
And record_details_by_id raises `ResourceNotFound` is captured above this exception.
Is this needed now? Then what scenario?